From 4e0f406df2399be8984a6b774669ec1181153a37 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Mon, 5 Jan 2015 10:34:13 +0200
Subject: [PATCH 28/59] ARM: i.MX6: ASoC: fix build warnings and update
 includes

Fix build warnings and update includes.

sound/soc/fsl/imx-wm8731.c: In function 'imx_hifi_hw_params_slv_mode':
sound/soc/fsl/imx-wm8731.c:357:3: warning: format '%u' expects type 'unsigned int', but argument 2 has type 'long int'
sound/soc/fsl/imx-wm8731.c: In function 'imx_hifi_hw_params_mst_mode':
sound/soc/fsl/imx-wm8731.c:414:3: warning: format '%u' expects type 'unsigned int', but argument 2 has type 'long int'

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 sound/soc/fsl/imx-wm8731.c |   13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/sound/soc/fsl/imx-wm8731.c b/sound/soc/fsl/imx-wm8731.c
index 72b75ad..c0833cf 100644
--- a/sound/soc/fsl/imx-wm8731.c
+++ b/sound/soc/fsl/imx-wm8731.c
@@ -14,17 +14,12 @@
  */
 
 #include <linux/module.h>
+#include <linux/of.h>
 #include <linux/of_platform.h>
 #include <linux/of_i2c.h>
-#include <linux/of_gpio.h>
-#include <linux/slab.h>
-#include <linux/gpio.h>
 #include <linux/clk.h>
 #include <sound/soc.h>
-#include <sound/jack.h>
 #include <sound/pcm_params.h>
-#include <sound/soc-dapm.h>
-#include <linux/pinctrl/consumer.h>
 
 #include "../codecs/wm8731.h"
 #include "imx-audmux.h"
@@ -344,7 +339,7 @@ static int imx_hifi_hw_params_slv_mode(struct snd_pcm_substream *substream,
 				     SND_SOC_CLOCK_IN);
 
 	if (ret < 0) {
-		pr_err("Failed to set codec master clock to %u: %d \n",
+		pr_err("Failed to set codec master clock to %lu: %d \n",
 		       data->sysclk, ret);
 		return ret;
 	}
@@ -401,7 +396,7 @@ static int imx_hifi_hw_params_mst_mode(struct snd_pcm_substream *substream,
 				     SND_SOC_CLOCK_IN);
 
 	if (ret < 0) {
-		pr_err("Failed to set codec master clock to %u: %d \n",
+		pr_err("Failed to set codec master clock to %lu: %d \n",
 		       data->sysclk, ret);
 		return ret;
 	}
@@ -555,7 +550,7 @@ static int imx_wm8731_probe(struct platform_device *pdev)
 	}
         
 	codec_dev = of_find_i2c_device_by_node(codec_np);
-	if (!codec_dev || !codec_dev->driver) {
+	if (!codec_dev) {
 		dev_err(&pdev->dev, "failed to find codec platform device\n");
 		ret = -EINVAL;
 		goto fail;
-- 
1.7.9.5

